viewing_party/party.py

```diff
@@ -1,23 +1,173 @@
+
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    
+    if title and genre and rating:
+        movie_dict = {}
+        movie_dict["title"] = title
+        movie_dict["genre"] = genre
+        movie_dict["rating"] = rating
+    else:
+        return None
+    
+    return movie_dict
+
+
+def add_to_watched(user_data, movie):
+
+    user_data["watched"].append(movie)
+
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+
+    user_data["watchlist"].append(movie)
+
+    return user_data
+
+
+def watch_movie(user_data, title):
+
+#loop through the dictionaries in the list user_data["watchlist"]
+#if the dictionary key == "title", check if the value == title (parameter)
+#if the value == title, save the dictionary to user_data["watched"]. Then delete the dictionary from user_data["watchlist"]
+
+
+    for dict in user_data["watchlist"]:
+        if dict["title"] == title:
+            user_data["watched"].append(dict)
+            user_data["watchlist"].remove(dict)

```

Comment:
Right now we continue iterating over the watchlist after we remove an item from it. In most cases this isn't a big deal, but say we were dealing with lists of thousands of movies. We would probably want to exit the loop right after we perform the operation we want, rather than continuing to loop over potentially hundreds of more titles! How would we be able to halt execution once we performed our swap?

Another consideration: we are removing a value from a list we are actively iterating over, so after the first removal, we skip a value because the way iterators work (recall our discussion on the "danger zone" from the roundtable on "Iterating over Data"). Given the data for our tests, we know this isn't a concern for this application, but what if we were working with data that had duplicate titles?

Reply:
How would we be able to halt execution once we performed our swap?: 
One option would be to put "break" after line 41 in the if statement. Another option could be to move the return statement up into the if statement. But I believe both of these options would only work if we were only performing the operation for one movie.

What if we were working with data that had duplicate titles?:
My first thoughts are to make a copy of user_data["watchlist"] to iterate over in the for loop, and only make the for loop for appending. The for loop would iterate over the copy_watchlist, and append to user_data["watched"] in the same way. I could make a second for loop that is the essentially the same, except it removes. I am thinking this would solve the "danger zone" and account for duplicates but I have a feeling there may be a better implementation out there.